tests/repl_rt_sink_rebalance.erl

```diff
@@ -0,0 +1,137 @@
+%% -------------------------------------------------------------------
+%%
+%% Copyright (c) 2013 Basho Technologies, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+%%
+
+-module(repl_rt_sink_rebalance).
+-behaviour(riak_test).
+-export([confirm/0]).
+-include_lib("eunit/include/eunit.hrl").
+
+confirm() ->
+    NumNodes = rt_config:get(num_nodes, 4),
+
+    lager:info("Deploy ~p nodes", [NumNodes]),
+    Conf = [
+            {riak_repl,
+             [
+              %% turn off fullsync
+              {fullsync_on_connect, false},
+              {fullsync_interval, disabled},
+              {realtime_connection_rebalance_max_delay_secs, 10}
+             ]}
+    ],
+
+    Nodes = rt:deploy_nodes(NumNodes, Conf, [riak_kv, riak_repl]),
+    {ANodes, BNodes} = lists:split(2, Nodes),
+
+    %% Set up to grep logs to verify messages
+    [rt:setup_log_capture(Node) || Node <- ANodes],
+
+    lager:info("ANodes: ~p", [ANodes]),
+    lager:info("BNodes: ~p", [BNodes]),
+
+    lager:info("Build cluster A"),
+    repl_util:make_cluster(ANodes),
+
+    lager:info("Build cluster B"),
+    repl_util:make_cluster(BNodes),
+
+    lager:info("waiting for leader to converge on cluster A"),
+    ?assertEqual(ok, repl_util:wait_until_leader_converge(ANodes)),
+    AFirst = hd(ANodes),
+
+    lager:info("waiting for leader to converge on cluster B"),
+    ?assertEqual(ok, repl_util:wait_until_leader_converge(BNodes)),
+    BFirst = hd(BNodes),
+
+    lager:info("Naming A"),
+    repl_util:name_cluster(AFirst, "A"),
+
+    lager:info("Naming B"),
+    repl_util:name_cluster(BFirst, "B"),
+
+    connect_clusters(AFirst, BFirst),
+
+    enable_rt(AFirst, ANodes),
+
+    [?assertEqual(ok,verify_connectivity(Node)) || Node <- ANodes],
+    [?assertEqual(ok,verify_sinks(Node)) || Node <- BNodes],
+
+    rt:stop(BFirst),
+    ?assertEqual(ok, rt:wait_until_unpingable(BFirst)),
+    lager:info("Let sleep for 10s"),
+    timer:sleep(10000),
+    rt:start(BFirst),
+    ?assertEqual(ok, rt:wait_until_nodes_ready([BFirst])),
+    ?assertEqual(ok, rt:wait_until_all_members(BNodes)),
+    ?assertEqual(ok, rt:wait_until_ring_converged(BNodes)),
+    timer:sleep(10000),
```

Comment:
So the issue here is we run verify_sinks too fast and look for the connection_manager before it exists resulting in a badrpc for reason noproc. This crashes the test. I should have changed verify sinks to also wait until the cluster_manager exists instead of the sleep but I knew that this would work while I was testing.